### DISCUSSION TITLE
Runtime components libraries update

### DISCUSSION THREAD

**Discussion by ScorpioNormal** [shreyashsaitwal/rush-cli#56] [General]
Rush uses **runtime.jar** and **runtime-sources.jar** archives to compile extensions. However, the content of the runtime components has not been updated since 2021. And some new class interaction features are not available for extensions. How to get new versions of these libraries?

**Comment by shreyashsaitwal** [comment]
Hi,

You can use the latest version of AI2 component classes in Rush by using the `AndroidRuntime.jar` that you get after building the AI2 sources. You can find this JAR in the `components/build` directory after building the components. Just place the JAR in the `deps` directory without listing it `deps` field in `rush.yml`. This prevents the JAR from being bundled with your extension while still allowing you to use its classes.

You can also use the ready-made `AndroidRuntime.jar` from MIT's extension-template's [dependencies](https://github.com/mit-cml/extension-deps/tree/78dc0a61b5f437ff6b54dc3998921fa599bea20f), but depending on when it was last updated, it might not contain all the latest changes.

**Comment by ScorpioNormal** [comment]
Hi!
Thanks for the answer! I tried both ways, but they don't work:
[cid:10b910d9-c778-4589-8e7f-54dbbd6b700e]
 I wanted to use a Canvas.getBitmap() new function in my extension. But Rush didn’t detect it.
I assume that Rush uses the libraries from this folder as priority:

[cid:bc2e79cb-65c8-4f57-9f84-f4e93a113f7b]
I compiled my extension the traditional way and it works.
However, in the future I would like to use Rush.
Good luck!
Dmitry

________________________________
De: Shreyash Saitwal ***@***.***>
Enviado: martes, 5 de diciembre de 2023 01:56 p. m.
Para: shreyashsaitwal/rush-cli ***@***.***>
Cc: ScorpioNormal ***@***.***>; Author ***@***.***>
Asunto: Re: [shreyashsaitwal/rush-cli] Runtime components libraries update (Discussion #56)


Hi,

You can use the latest version of AI2 component classes in Rush by using the AndroidRuntime.jar that you get after building the AI2 sources. You can find this JAR in the components/build directory after building the components. Just place the JAR in the deps directory without listing it deps field in rush.yml. This prevents the JAR from being bundled with your extension while still allowing you to use its classes.

You can also use the ready-made AndroidRuntime.jar from MIT's extension-template's dependencies<https://github.com/mit-cml/extension-deps/tree/78dc0a61b5f437ff6b54dc3998921fa599bea20f>, but depending on when it was last updated, it might not contain all the latest changes.
